src/NuGet.Core/NuGet.ProjectModel/TargetFrameworkInformation.cs

```diff
@@ -64,6 +70,7 @@ public override int GetHashCode()
             hashCode.AddSequence(DownloadDependencies);
             hashCode.AddSequence(FrameworkReferences);
             hashCode.AddObject(RuntimeIdentifierGraphPath);
+            hashCode.AddSequence(CentralVersionDependencies);
```

Comment:
Did you verify that AddSequence hanldes dictionaries correctly, because they are not sorted?

Reply:
I added unit tests. All the tests are ok.